packages/razorsharp/src/code/blade/utils/spacing.ts

```diff
@@ -0,0 +1,68 @@
+import { SPACING_TO_TOKEN } from '../constants/spacing';
+
+export const getTokenFromSpacingValue = (space: number): string => {
+  // check if there is a token for the exact value
+  if (space in SPACING_TO_TOKEN) {
+    return `spacing.${SPACING_TO_TOKEN[space as keyof typeof SPACING_TO_TOKEN]}`;
+  }
+
+  // threshold for the nearest value search
+  const THRESHOLD = 2;
+  for (const spacingValue in SPACING_TO_TOKEN) {
+    if (Object.prototype.hasOwnProperty.call(SPACING_TO_TOKEN, spacingValue)) {
+      const token = SPACING_TO_TOKEN[+spacingValue as keyof typeof SPACING_TO_TOKEN];
+
+      if (Math.abs(+spacingValue - space) <= THRESHOLD) {
```

Comment:
TODO: warn user that we are matching a nearest value, this means that the spacing value wasn't properly used in design side.